BSP/Simulator/Src/BSP_WDTimer.c

```diff
@@ -47,38 +27,8 @@ void BSP_WDTimer_Init(void) {
  * @return  true if a reset occurred previously, false if system started up normally.
  */
 bool BSP_WDTimer_DidSystemReset(void){
-    if(didSystemReset == true){
-        didSystemReset = false;      //clear flag
-        return true;
-    }else{
-        return didSystemReset;
-    }
-}
-
-//checks if system reset flag
-void CheckReset(void){
-    char resetValue;
-    FILE* fp = fopen(file, "r+");       //check if python script left system reset flag ('b')
-    if(!fp) {
-        perror(WDTIMER_CSV_FILE);
-        exit(EXIT_FAILURE);
-    }
-    int fno = fileno(fp);
-    flock(fno, LOCK_EX);
-
-    fseek(fp, 2, SEEK_SET);
-    resetValue = fgetc(fp);
-    if(resetValue == 'b'){
-        didSystemReset = true;
-    }else{
-        didSystemReset = false;
-    }
-    flock(fno, LOCK_UN);
-    fclose(fp);
-    hasStarted = false; 
-
-    
-    
+    // TODO
```

Comment:
a suggestion: declare variable `volatile bool hasReset = false`
and have `hasReset = true` in `BSP_WDTimer_Reset()`. then replace this return false with `return hasReset`

Reply:
The watchdog timer is a hardware timer that will trip the system if it counts down to 0. A flag is set and can be read when the system resets. This cannot be simulated by the simulator since we do not have a watchdog timer.

Reply:
oof